src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -139,7 +139,12 @@ static Slice build(final Settings settings, final Authentication auth,
         final Permissions permissions = new LoggingPermissions(
             cfg.permissions().orElse(Permissions.FREE)
         );
-        final Pattern prefix = new PathPattern(settings).pattern();
+        final Pattern prefix;
+        if (cfg.port() == -1) {
+            prefix = new PathPattern(settings).pattern();
+        } else {
+            prefix = Pattern.compile("()(/.*)?");
+        }
```

Comment:
@Sammers21 I am not sure that it's okey to not serve only repositories without `port` on default port. Instead we should serve repositories with host on their host and on default host by name as well